src/components/icon-button/IconButton.styles.ts

```diff
@@ -0,0 +1,329 @@
+import styled, { css } from 'styled-components';
+import { FlattenSimpleInterpolation } from 'styled-components/ts3.6';
+
+import { TIconButtonSize, IconButtonLayer, IconButtonState } from './IconButton.types';
+import {
+    ICON_BUTTON_SIZES,
+    ICON_BUTTON_DIMENSIONS,
+    ICON_BUTTON_PADDING,
+    ICON_BUTTON_SPACING,
+} from './IconButton.constants';
+
+function generateSizingStyles(size: TIconButtonSize): FlattenSimpleInterpolation {
+    return css`
+        &[data-size='${size}'] {
+            --horizontal-padding: ${ICON_BUTTON_PADDING[size]}px;
+            --size: ${ICON_BUTTON_DIMENSIONS[size]}px;
+            --child-spacing: ${ICON_BUTTON_SPACING[size]}px;
+        }
+    `;
+}
+
+function getRBGColour({ colour, opacity }: IconButtonLayer): string {
+    if (opacity && opacity >= 0 && opacity < 1) {
+        return `rgba(var(${colour}), ${opacity})`;
+    }
+
+    return `rgb(var(${colour}))`;
+}
+
+function getLayerColours(layer: IconButtonLayer): string {
+    return `--colour-${layer.name}: ${getRBGColour(layer)};`;
+}
+
+function generateStateStyles(stateStyles: IconButtonState[]): FlattenSimpleInterpolation {
+    return css`
+        ${stateStyles
+            .map(({ selector, layers }: IconButtonState) => {
+                if (selector === 'base') {
+                    return `
+                        ${layers.map(getLayerColours).join('\n')}
+                    `;
+                }
+
+                return `
+                    &:${selector} {
+                        ${layers.map(getLayerColours).join('\n')}
+                    }
+                `;
+            })
+            .join('\n')};
+    `;
+}
+
+const StyledIconButton = styled.button`
```

Comment:
we should probably add the `shouldForwardProp` config to each of our styled components

Reply:
@michelengelen, can you pleas add?